### PR TITLE
Polish unit tests of defaultpreemptio plugin

### PR DESCRIPTION
**Pull request by Huang-Wei**
**What type of PR is this?**

/kind cleanup
/sig scheduling

**What this PR does / why we need it**:

The PR polishes preemption unit tests by:

1. Use a standard *Wrapper to construct pod/node.
1. Remove/simplify unnecessary logic in main testing blocks.

Additionally, this PR also adds a test to cover `PostFilter` of defaultpreemption plugin.

(After this PR, there will be a final PR to expose/rename some internal functions to end this series.)

**Which issue(s) this PR fixes**:

Part of #91038

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by chendave — COMMENTED** [review]
haven't went through the details, but seem like there is a typo in the commit message :)
s/defaultpreemptio plugin/defaultpreemption plugin

**Comment by Huang-Wei** [comment]
@chendave yup, thanks for pointing it out. I will amend the commit msg along with other fixups.

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go [hunk 1]
```diff
@@ -226,422 +97,569 @@ func mergeObjs(pod *v1.Pod, pods []*v1.Pod) []runtime.Object {
...
+				},
+			},
 			expectedNumFilterCalled: 5,
 		},
 		{
 			name: "mixed priority pods are preempted, pick later StartTime one when priorities are equal",
 			registerPlugins: []st.RegisterPluginFunc{
-				st.RegisterQueueSortPlugin(queuesort.Name, queuesort.New),
 				st.RegisterPluginAsExtensions(noderesources.FitName, noderesources.NewFit, "Filter", "PreFilter"),
-				st.RegisterBindPlugin(defaultbinder.Name, defaultbinder.New),
```
Comment (ahg-g) [inline pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go:357]: why did we need to register bind and queue plugins before?

**Review by ahg-g — COMMENTED** [review]
would have been easier to review if it was broken into multiple PRs: 1) renaming, 2) wrapping objects 3) adding new test etc.

I did take a look and tried to verify, but can you please confirm that non of the tests changed its expectation? otherwise looks good to me.

**Comment by Huang-Wei** [comment]
@ahg-g Thanks for the review. Apologizing for combining several things together. And yes, there are 3 kinds of changes:

1. renaming - in some places, the word "machine" is renamed to "node" as "machine" always reminds me of a bare metal...
1. wrapping objects - the added wrapper functions are just for simplifying the testing code
1. new test - `TestPostFilter` is newly added, it's a new addition.

So overall, all old testing artifacts are 100% kept and their functions remain unchanged. The addition `TestPostFilter` to cover the path of `PostFilter()`.

File: pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go (hunk 1 above)
Comment (Huang-Wei) [reply pkg/scheduler/framework/plugins/defaultpreemption/default_preemption_test.go:357]: As `NewFramework` would report missing mandatory plugins (queuesort or bind):

https://github.com/kubernetes/kubernetes/blob/c53ce48d48372c30053a9b67c6d1dee237b5cd69/pkg/scheduler/framework/runtime/framework.go#L307-L315

**Pull request merged**